tools/src/main/java/org/apache/kafka/tools/reassign/ReassignPartitionsCommand.java

```diff
@@ -0,0 +1,1501 @@
+                } else {
+                    String currentReplicaStr = state.currentReplicas.stream().map(String::valueOf).collect(Collectors.joining(","));
+                    String targetReplicaStr = state.targetReplicas.stream().map(String::valueOf).collect(Collectors.joining(","));
+
+                    bld.add("There is no active reassignment of partition " + topicPartition + ", " +
+                        "but replica set is " + currentReplicaStr + " rather than " +
+                        targetReplicaStr + ".");
+                }
+            } else {
+                bld.add(String.format("Reassignment of partition %s is still in progress.", topicPartition));
+            }
+        });
+        return bld.stream().collect(Collectors.joining(System.lineSeparator()));
+    }
+
+    /**
+     * Find the state of the specified partition reassignments.
+     *
+     * @param adminClient          The Admin client to use.
+     * @param targetReassignments  The reassignments we want to learn about.
+     *
+     * @return                     A tuple containing the reassignment states for each topic
+     *                             partition, plus whether there are any ongoing reassignments.
+     */
+    static Tuple2<Map<TopicPartition, PartitionReassignmentState>, Boolean> findPartitionReassignmentStates(Admin adminClient,
+                                                                                                            List<Tuple2<TopicPartition, List<Integer>>> targetReassignments) throws ExecutionException, InterruptedException {
+        Map<TopicPartition, PartitionReassignment> currentReassignments = adminClient.
+            listPartitionReassignments().reassignments().get();
+
+        List<Tuple2<TopicPartition, List<Integer>>> foundReassignments = new ArrayList<>();
+        List<Tuple2<TopicPartition, List<Integer>>> notFoundReassignments = new ArrayList<>();
+
+        targetReassignments.forEach(reassignment -> {
+            if (currentReassignments.containsKey(reassignment.v1))
+                foundReassignments.add(reassignment);
+            else
+                notFoundReassignments.add(reassignment);
+        });
+
+        List<Tuple2<TopicPartition, PartitionReassignmentState>> foundResults = foundReassignments.stream().map(e -> {
+            TopicPartition part = e.v1;
+            List<Integer> targetReplicas = e.v2;
+            return new Tuple2<>(part,
+                new PartitionReassignmentState(
+                    currentReassignments.get(part).replicas(),
+                    targetReplicas,
+                    false));
+        }).collect(Collectors.toList());
+
+        Set<String> topicNamesToLookUp = new HashSet<>();
+        notFoundReassignments.forEach(e -> {
+            TopicPartition part = e.v1;
+            if (!currentReassignments.containsKey(part))
+                topicNamesToLookUp.add(part.topic());
+        });
```

Comment:
What about using simple stream API here:
```suggestion
        Set<String> topicNamesToLookUp0 = notFoundReassignments.stream()
            .map(e -> e.v1)
            .filter(part -> !currentReassignments.containsKey(part))
            .map(TopicPartition::topic)
            .collect(Collectors.toSet());
```

Reply:
Greate suggestion. Thanks.